### PR TITLE
datacube doc updates

### PR DESCRIPTION
**Pull request by jonathom**
new datacubes updates, as discussed. Do we want another graphic explaining the concept of additional dimensions (CRS, sensors)? I think that could be beneficial, but maybe not needed.

### CODE REVIEW

File: documentation/1.0/glossary.md [hunk 1]
```diff
@@ -33,137 +33,7 @@ In openEO, a back-end offers a set of collections to be processed. All collectio
...
-
-On such a _crs-dimensioned data cube_, several operations make perfect sense, such as `apply` or `reduce_dimension` on spectral and/or temporal dimensions. A simple reduction over the `crs` dimension, using _sum_ or _mean_ would typically not make sense. The "reduction" (removal) of the `crs` dimension that is meaningful involves the resampling/warping of all sub-cubes for the `crs` dimension to a single, common target coordinate reference system.
+A spatiotemporal datacube is a multidimensional array with one or more spatial or temporal dimensions. In the EO domain, it is common to be implicit about the temporal dimension and just refer to them as spatial datacubes in short. Special cases are raster and vector datacubes. Learn more about them in the [datacube documentation](https://openeo.org/documentation/1.0/datacubes.html).
```
Comment (m-mohr) [inline documentation/1.0/glossary.md:36]:
```suggestion
A spatiotemporal data cube is a multidimensional array with one or more spatial or temporal dimensions. In the EO domain, it is common to be implicit about the temporal dimension and just refer to them as spatial datacubes in short. Special cases are raster and vector data cubes. Learn more about data cubes in the [data cube documentation](https://openeo.org/documentation/1.0/datacubes.html).
```

File: documentation/1.0/datacubes.md [hunk 2]
```diff
@@ -37,6 +37,14 @@ Here is an overview of the dimensions contained in our example datacube above:
 | 3 | `bands`          | `blue`, `green`, `red`, `nir` | 4 bands |
 | 4 | `t`              | `2020-10-01`, `2020-10-13`, `2020-10-25` | 12 days |
 
+Dimension labels are either of type number or string, including all sub types such as integers or temporal strings. Additional data types may be allowed in the future. Dimensions with a natural/inherent label order (with a reference system this is each spatial and temporal dimension) are always sorted. Other dimensions where there is no inherent order, including bands, have the order in which they have been defined in metadata or processes (e.g. [`filter_bands`](https://processes.openeo.org/#filter_bands)) and new labels are added at the end.
+
+OpenEO datacubes contain scalar pixels, with all other information stored in dimensions (e.g. coordinates or timestamps). Additional attributes (e.g. "CRS" or "Sensor") can also be turned into dimensions. Be advised that in such a case, the uniqueness of pixel coordinates is affected. When usually, `(x, y)` refers to a unique location, that changes to `(x, y, CRS)` when `(x, y)` values are reused in other coordinate reference systems (e.g. two neighbouring UTM zones).
```
Comment (m-mohr) [inline documentation/1.0/datacubes.md:42]:
```suggestion
OpenEO datacubes contain scalar values (e.g. strings, numbers or boolean values), with all other associated attributes stored in dimensions (e.g. coordinates or timestamps). Attributes such as the CRS or the sensor can also be turned into dimensions. Be advised that in such a case, the uniqueness of pixel coordinates may be affected. When usually, `(x, y)` refers to a unique location, that changes to `(x, y, CRS)` when `(x, y)` values are reused in other coordinate reference systems (e.g. two neighboring UTM zones).
```

File: documentation/1.0/datacubes.md [hunk 3]
```diff
@@ -37,6 +37,14 @@ Here is an overview of the dimensions contained in our example datacube above:
...
+
+::: tip Be Careful with Data Types
+As stated above, datacubes only contain scalar pixels. However, data types are handled by each backend individually. To avoid unhandled data types, steer clear of changing or mixing data types in datacube functions. Do not rely on a backend to automatically do conversions for you, unless explicitly stated by that backend.
```
Comment (m-mohr) [inline documentation/1.0/datacubes.md:45]:
```suggestion
As stated above, datacubes only contain scalar values. However, implementations may differ in their ability to handle or convert them. Implementations may also not allow mixing data types in a data cube. For example, returning a boolean value for a reducer on a numerical data cube may result in an error on some back-ends. The recommendation is to not change the data type of values in a data cube unless the back-end supports it explicitly.
```

File: documentation/1.0/datacubes.md [hunk 4]
```diff
@@ -37,6 +37,14 @@ Here is an overview of the dimensions contained in our example datacube above:
 | 3 | `bands`          | `blue`, `green`, `red`, `nir` | 4 bands |
 | 4 | `t`              | `2020-10-01`, `2020-10-13`, `2020-10-25` | 12 days |
 
+Dimension labels are either of type number or string, including all sub types such as integers or temporal strings. Additional data types may be allowed in the future. Dimensions with a natural/inherent label order (with a reference system this is each spatial and temporal dimension) are always sorted. Other dimensions where there is no inherent order, including bands, have the order in which they have been defined in metadata or processes (e.g. [`filter_bands`](https://processes.openeo.org/#filter_bands)) and new labels are added at the end.
```
[the remark this answers is not in the document]
Comment (jonathom) [reply documentation/1.0/datacubes.md:40]:
```suggestion
Dimension labels are either numerical or text (also known as "strings"), which for example also includes textual representations of timestamps. Dimensions with a natural/inherent order are always sorted. This are usually all spatial and temporal dimensions. Dimensions without inherent order, for example `bands` in openEO, retain the order in which they have been defined in metadata or processes (e.g. [`filter_bands`](https://processes.openeo.org/#filter_bands)) and new labels are added at the end.
```
Would change like so.
What is meant by ".. new labels are added at the end."? Before, label order is described. How does that connect to labels being added at the end - what end?

File: documentation/1.0/datacubes.md (hunk 3 above)
Comment (jonathom) [reply documentation/1.0/datacubes.md:45]:
```suggestion
As stated above, datacubes only contain scalar values. However, implementations may differ in their ability to handle or convert them. Implementations may also not allow mixing data types in a datacube. For example, returning a boolean value for a reducer on a numerical data cube may result in an error on some back-ends. The recommendation is to not change the data type of values in a datacube unless the back-end supports it explicitly.
```
data cube -> datacube

Comment (jonathom) [reply documentation/1.0/datacubes.md:45]:
```suggestion
As stated above, datacubes only contain scalar values. However, implementations may differ in their ability to handle or convert them. Implementations may also not allow mixing data types in a datacube. For example, returning a boolean value for a reducer on a numerical datacube may result in an error on some back-ends. The recommendation is to not change the data type of values in a datacube unless the back-end supports it explicitly.
```

File: documentation/1.0/datacubes.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (m-mohr) [reply documentation/1.0/datacubes.md:40]: Sounds good. 

Labels are like an array which have the order as defined, but new labels are added to the end of the array.

File: documentation/1.0/datacubes.md (hunk 3 above)
Comment (m-mohr) [reply documentation/1.0/datacubes.md]: Looks good.

**Comment by m-mohr** [comment]
There are three additional "data cube" instead of "datacube". Could you also replace those? Especially the heading... :-)

File: documentation/1.0/datacubes.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (jonathom) [reply documentation/1.0/datacubes.md:40]:
```suggestion
Dimension labels are either numerical or text (also known as "strings"), which for example also includes textual representations of timestamps. Dimensions with a natural/inherent order are always sorted. This are usually all spatial and temporal dimensions. Dimensions without inherent order, for example `bands` in openEO, retain the order in which they have been defined in metadata or processes (e.g. [`filter_bands`](https://processes.openeo.org/#filter_bands)). New labels are always added at the end of the label array.
```
would make an own sentence then.

[the remark this answers is not in the document]
Comment (m-mohr) [reply documentation/1.0/datacubes.md:40]: Wait, this only applies to those bands without inherent order, so the "always" might be confusing. For temporal/spatial (with inherent order) they are added in the right spot with ordering. That's why it was in one sentence.

[the remark this answers is not in the document]
Comment (m-mohr) [reply documentation/1.0/datacubes.md:40]: Maybe:

> For those other dimensions, new labels will be added after the last label.

[the remark this answers is not in the document]
Comment (jonathom) [reply documentation/1.0/datacubes.md:40]:
```suggestion
Dimension labels are either numerical or text (also known as "strings"), which for example also includes textual representations of timestamps. Dimensions with a natural/inherent order are always sorted. This are usually all spatial and temporal dimensions. Dimensions without inherent order, for example `bands` in openEO, retain the order in which they have been defined in metadata or processes (e.g. [`filter_bands`](https://processes.openeo.org/#filter_bands)), with new labels added at the end of the label array.
```

[the remark this answers is not in the document]
Comment (m-mohr) [reply documentation/1.0/datacubes.md:40]: The array was just for descriptive purposes, it's not necessarily an array internally. So would prefer to not say "label array", which is not used anywhere else.

So maybe:

> , with new labels being added after the existing labels.

[the remark this answers is not in the document]
Comment (jonathom) [reply documentation/1.0/datacubes.md:40]: > with new labels simply appended to the existing labels

or
> with new labels simply added after the existing labels

?

[the remark this answers is not in the document]
Comment (jonathom) [reply documentation/1.0/datacubes.md:40]: adding "simply" and fiddling with this to come up with a sentence that I would've understood :)

[the remark this answers is not in the document]
Comment (m-mohr) [reply documentation/1.0/datacubes.md:40]:
```suggestion
Dimension labels are either numerical or text (also known as "strings"), which for example also includes textual representations of timestamps. Dimensions with a natural/inherent order are always sorted. These are usually all spatial and temporal dimensions. Dimensions without inherent order, `bands` in openEO for example, retain the order in which they have been defined in metadata or processes (e.g. through [`filter_bands`](https://processes.openeo.org/#filter_bands)), with new labels simply being appended to the existing labels.
```

[the remark this answers is not in the document]
Comment (m-mohr) [reply documentation/1.0/datacubes.md:40]: Some additional changes above for better grammar, I think. Integrating your latest proposal, I like the "appended"

**Pull request merged**